### PR TITLE
TaskDescription.waitForAllTaskTermination() on waitOnCommand()

### PR DESCRIPTION
**Pull request by norio-nomura**
*This is rebased version of #496.*

Fix segmentation fault on termination of Carthage.

Segmentation fault on termination of Carthage 0.7.4 beta-2 happens following scenario:
1. launchTask's thread call `sendCompleted()` and kick main thread
2a. main thread wake up and call `exit()`
2b. launchTask's thread continue disposing
3. main thread cause segmentation fault.

Waiting `completed` from launchTask's thread is not enough for fixing the segmentation fault.
It needs waiting actual termination of launchTask's thread

This PR depends on https://github.com/Carthage/ReactiveTask/pull/24

https://github.com/Carthage/Carthage/pull/474#issuecomment-107949958

### CODE REVIEW

File: Source/carthage/Extensions.swift [hunk 1]
```diff
@@ -108,6 +109,7 @@ internal func waitOnCommand<T>(producer: SignalProducer<T, CommandError>) -> Res
 		|> then(SignalProducer<(), CommandError>.empty)
 		|> wait
 	
+	TaskDescription.waitForAllTaskTermination()
```
Comment (robrix) [inline Source/carthage/Extensions.swift]: Should this be finer-grained, e.g. waiting per-task when they’re executed?

Comment (norio-nomura) [inline Source/carthage/Extensions.swift]: I can't determined where I should place `dispatch_group_wait()` on another place.
RAC's `completed` event timing is not fit for place it.

Comment (norio-nomura) [inline Source/carthage/Extensions.swift]: The segmentation fault happens while disposing after `completed` event from launch task's thread and main thread calling `exit()` concurrently.
Waiting per-task will need waiting some RAC event. That will be same what just current implementation is doing.
I think fixing the segmentation fault needs doing outside of RAC event, and waiting per-task without RAC event may be too complicated unnecessarily.

Comment (robrix) [inline Source/carthage/Extensions.swift]: Okay, thanks for looking into this!

**Comment by robrix** [comment]
I’ve merged [the dependency](https://github.com/Carthage/ReactiveTask/pull/24) and have [a beta tag](https://github.com/Carthage/ReactiveTask/releases/tag/0.7-beta1) of ReactiveTask to include it.

Could you please update the `Cartfile` & submodules to pin to that tag?

**Comment by robrix** [comment]
Once that’s done I think we can merge and release a new beta of Carthage. I may not be aesthetically content with waiting for all child processes like this, but I can’t argue with its results.

**Comment by norio-nomura** [comment]
I've pushed the commit updating the `Cartfile` & submodules to pin to `0.7-beta1`

**Comment by robrix** [comment]
Thanks very much!

**Comment by norio-nomura** [comment]
You're welcome. :bow:

**Pull request merged**